t/core/config_etcd.t

```diff
@@ -443,3 +443,81 @@ qr/readdir key: fake res: \{.*"nodes":\[\{.*"value":\["bar"\].*\}\].*\}/
 --- wait: 1
 --- no_error_log
 [error]
+
+
+
+=== TEST 13: test route with special character "-"
```

Comment:
Add all tests in one "TEST" because we need to ensure nginx not be reloaded to cover this bug.

Reply:
Just to make sure, the bug can be reproduced even nginx is reloaded, right?

Reply:
Yes, even if nginx is reloaded it will also reproduce this bug as long as there are new events come.